### PR TITLE
feat: adds xy plotting methods with tests and docs

### PR DESCRIPTION
**Pull request by MartinWohlan**
This PR contains the following features:

- xy.plot() function which can plot positions and trajectories:
   - vis.positions.py which contains the actual plotting functions that xy.plot() uses.
   - vis.utils.py which contains a decorator to check if matplotlib.axes is given and creates one if not.
- Tests for xy.plot() and vis.positions
- Docs with notes and examples for pitch.plot() and xy.plot()
- Additionally some bugs, typos and formatting adjustments

### CODE REVIEW

File: floodlight/vis/positions.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    Parameters
+    ----------
+    xy: floodlight.core.xy.XY
+        Spatio-temporal data fragment object that contains the positions
+        (XY.xy: np.ndarray) and functions for manipulation.
+    frame: int
+        Frame in the postions-array (XY.xy) for which the positions are plotted.
+    ball: bool
+        If set to False marker="o". Else marker="."
```
Comment (draabe) [inline floodlight/vis/positions.py:21]:
```suggestion
    ball: bool
        Boolean indicating whether this object is storing ball data. If set to False marker="o", else marker=".".
```

File: floodlight/vis/positions.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import matplotlib
+
+from floodlight.vis.utils import check_axes_given
```
Comment (draabe) [inline floodlight/vis/positions.py:3]:
```suggestion
from floodlight import XY
from floodlight.vis.utils import check_axes_given
```

File: floodlight/vis/positions.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+
+@check_axes_given
+def plot_positions(
+    xy, frame: int, ball: bool, ax: matplotlib.axes, **kwargs
```
Comment (draabe) [inline floodlight/vis/positions.py:8]:
```suggestion
    xy: XY, frame: int, ball: bool, ax: matplotlib.axes, **kwargs
```

File: floodlight/vis/positions.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+) -> matplotlib.axes:
+    """Scatter plots positions for a given frame of the XY.xy array on a
+    matplotlib.axes.
```
Comment (draabe) [inline floodlight/vis/positions.py:11]:
```suggestion
    """Scatter plots positions for a given frame of an XY object on a
    matplotlib.axes.
```

File: floodlight/vis/positions.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    Parameters
+    ----------
+    xy: floodlight.core.xy.XY
+        Spatio-temporal data fragment object that contains the positions
+        (XY.xy: np.ndarray) and functions for manipulation.
```
Comment (draabe) [inline floodlight/vis/positions.py:17]:
```suggestion
        XY object containing spatiotemporal data to be plotted.
```

File: floodlight/vis/positions.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    frame: int
+        Frame in the postions-array (XY.xy) for which the positions are plotted.
```
Comment (draabe) [inline floodlight/vis/positions.py:19]:
```suggestion
        Number of frame to be plotted.
```

File: floodlight/vis/positions.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    **kwargs,
+) -> matplotlib.axes:
+    """Draws the trajectories of the XY.xy array for a given range of frames on a
+    matplotlib.axes.
+
+    Parameters
+    ----------
+    xy: floodlight.core.xy.XY
+        Spatio-temporal data fragment object that contains the positions
+        (XY.xy: np.ndarray) and functions for manipulation.
```
Comment (draabe) [inline floodlight/vis/positions.py:82]:
```suggestion
        XY object containing spatiotemporal data to be plotted.
```

File: floodlight/vis/positions.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    start_frame: int
+        First frame of the range of frames for which the trajectories are drawn.
+    end_frame: int
+        Last frame of the range of frames for which the trajectories are drawn.
```
Comment (draabe) [inline floodlight/vis/positions.py:86]:
```suggestion
        Closing frame of time interval to be plotted.
```

File: floodlight/vis/positions.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    ball: bool
+        If set to False marker="o". Else marker="."
```
Comment (draabe) [inline floodlight/vis/positions.py:88]:
```suggestion
    ball: bool
        Boolean indicating whether this object is storing ball data. If set to False marker="o", else marker=".".
```

File: floodlight/vis/utils.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def check_axes_given(func):
+    """Decorator function that checks if a matplotlib.axes is given as an argument.
+    Creates one if not.
+
+    Parameters
+    ----------
+    func:
+        Object of a function that needs a matplotlib.axes as an argument. If ax == None
```
Comment (draabe) [inline floodlight/vis/utils.py:11]:
```suggestion
        Function object that needs a matplotlib.axes as an argument. If ax == None
```

File: floodlight/vis/utils.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+        an axes is created an passed to the given function object as a keyworded
+        argument.
+
+    Returns
+    -------
+    func():
```
Comment (draabe) [inline floodlight/vis/utils.py:17]:
```suggestion
    func:
```

**Review by draabe — COMMENTED** [review]
I've added some cosmetics, please let me know if thatworks for you

File: floodlight/vis/positions.py (hunk 2 above)
Comment (MartinWohlan) [reply floodlight/vis/positions.py:3]: Leads to circular import error.

**Review by draabe — APPROVED** [review]

**Comment by draabe** [comment]
Awesome, thanks a lot!

**Pull request merged**